src/js/plugin.js

```diff
@@ -451,6 +451,16 @@ Plugin.BASE_PLUGIN_NAME = BASE_PLUGIN_NAME;
 
 Plugin.registerPlugin(BASE_PLUGIN_NAME, Plugin);
 
+/**
+ * Documented in player.js
+ *
+ * @ignore
+ */
+Player.prototype.usePlugin = function(name, plugin, ...options) {
+  Plugin.registerPlugin(name, plugin);
```

Comment:
I am against this as well. This method should *only* be a shorthand for calling the plugin manually.

When you register a plugin, it registers it with videojs, so, a plugin will be available to be initialized on all players and not just the player you called `usePlugin` on. For example:
```js
player.usePlugin('foo', function() {});

typeof player2.foo;
// function
```
That would be unexpected to me since I called `usePlugin` on `player`, why does `player2` have the option of initializing `foo`?

Reply:
Well, but having to register and then call the plugin, break the type inherence, that is one of main reason behind it... Is there any way to do this without registering plugin "globally"?

But I do agree, that registering on one player and accessing on the other would lead to confusion.